Bio/PDB/PDBList.py

```diff
@@ -35,18 +35,46 @@
 
 """Access the PDB over the internet (e.g. to download structures)."""
 
+from __future__ import annotations
 
+import collections
 import contextlib
+import dataclasses
 import ftplib
+import functools
 import gzip
 import os
 import re
 import shutil
+import socket
 import sys
+import time
 
 from urllib.request import urlopen
 from urllib.request import urlretrieve
 from urllib.request import urlcleanup
+from urllib.parse import urljoin
+from urllib.parse import urlunsplit
+
+from Bio.PDB.PDBExceptions import PDBException
+
+
+@dataclasses.dataclass
+class PDBServer:
+    """Protein Data Server class.
```

Comment:
Suggestion. Usually docstrings are a little more descriptive in the first line and have some description of the attributes/methods. For example, here I'd write 

```python
class PDBserver:
    """Dataclass to store parameters for connecting to PDB server mirrors.

        :param domain: base domain address of the server.
        :type domain: string

        :param pdb_directory: sub-domain or URL to the PDB data folder.
        :type domain: string
    """
```

Reply:
:+1: (minus vocabulary)